src/lib/gateway/gateway.ts

```diff
@@ -38,61 +34,63 @@ class Gateway extends GatewayBase {
     return this.makeRequest(GatewayServiceType.IS_ALIVE);
   }
 
-  public withdrawal(request: WithdrawRequest): Promise<Record<string, string>> {
-    return this._addTransaction(GatewayRequestType.WITHDRAWAL_REQUEST, request);
+  public withdrawal(
+    request: TransactionRequest
+  ): Promise<Record<string, string>> {
+    return this.addTransaction(GatewayRequestType.WITHDRAWAL_REQUEST, request);
   }
 
-  public deposit(request: DepositRequest): Promise<Record<string, string>> {
-    return this._addTransaction(GatewayRequestType.DEPOSIT_REQUEST, request);
+  public deposit(request: TransactionRequest): Promise<Record<string, string>> {
+    return this.addTransaction(GatewayRequestType.DEPOSIT_REQUEST, request);
   }
 
-  public mint(request: MintRequest): Promise<Record<string, string>> {
-    return this._addTransaction(GatewayRequestType.MINT_REQUEST, request);
+  public mint(request: TransactionRequest): Promise<Record<string, string>> {
+    return this.addTransaction(GatewayRequestType.MINT_REQUEST, request);
   }
 
   public settlement(
     request: SettlementRequest
   ): Promise<Record<string, string>> {
-    return this._addTransaction(GatewayRequestType.SETTLEMENT_REQUEST, request);
+    return this.addTransaction(GatewayRequestType.SETTLEMENT_REQUEST, request);
   }
 
   public transfer(request: TransferRequest): Promise<Record<string, string>> {
-    return this._addTransaction(GatewayRequestType.TRANSFER_REQUEST, request);
+    return this.addTransaction(GatewayRequestType.TRANSFER_REQUEST, request);
   }
 
   public conditionalTransfer(
     request: ConditionalTransferRequest
   ): Promise<Record<string, string>> {
-    return this._addTransaction(
+    return this.addTransaction(
       GatewayRequestType.CONDITIONAL_TRANSFER_REQUEST,
       request
     );
   }
 
   public fullWithdrawal(
-    request: FullWithdrawalRequest
+    request: TransactionRequest
   ): Promise<Record<string, string>> {
-    return this._addTransaction(
+    return this.addTransaction(
       GatewayRequestType.FULL_WITHDRAWAL_REQUEST,
       request
     );
   }
 
   public falseFullWithdrawal(
-    request: FalseFullWithdrawalRequest
+    request: TransactionRequest
   ): Promise<Record<string, string>> {
-    return this._addTransaction(
+    return this.addTransaction(
       GatewayRequestType.FALSE_FULL_WITHDRAWAL_REQUEST,
       request
     );
   }
 
-  private async _addTransaction(
+  private async addTransaction(
     type: GatewayRequestType,
```

Comment:
so why is the `type` not part of the `GatewayRequest` object?
do you expect a given request object to be used with different `type` values?

I can see they are all `TransactionRequest` objects, which is good IMO, so why not have the type as part of the object?

Reply:
I split it into different APIs `deposit, withdrawal, mint, etc...` in which I send the proper type, so it's internal handling and not something that should be passed from outside.

Reply:
Yes, I saw that.

You mean you don't want external users to set the type themselves?

Reply:
Yes, this was my thinking.